packages/data/addon/adapters/bard-facts-v2.ts

```diff
@@ -0,0 +1,406 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * Description: The adapter for the bard-facts request v2 model.
+ */
+
+import { assert } from '@ember/debug';
+import { inject as service } from '@ember/service';
+import { A as array } from '@ember/array';
+import { assign } from '@ember/polyfills';
+import EmberObject from '@ember/object';
+import { canonicalizeMetric, serializeParameters, getAliasedMetrics, canonicalizeAlias } from '../utils/metric';
+import { configHost } from '../utils/adapter';
+
+const SORT_DIRECTIONS = ['desc', 'asc'];
+
+type RequestOptions = {
+  clientId?: string;
+  customHeaders?: Dict<string>;
+  timeout?: number;
+  page?: number;
+  perPage?: number;
+  format?: string;
+  cache?: boolean;
+  queryParams?: Dict<string>;
+};
+type Query = RequestOptions & Dict<string | number | boolean>;
+type SortDirection = typeof SORT_DIRECTIONS[number];
+type ColumnType = 'metric' | 'dimension' | 'timeDimension';
+type Parameters = Dict<string>;
```

Comment:
Since parameters end up in the api request string, it makes since to me that their values should be strings. I'm not sure if there's a case where we specifically need a parameter value to be a number.